### PR TITLE
JavaScript And Ajax

### PR DESCRIPTION
**Pull request by rathoreprem**
This assignment based on Javascript abd Ajax Toolkit as well as Visualpage

### CODE REVIEW

File: JavaScript and Ajax/ClassDetailPage.page [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+   <form style="margin: 20px;" id="addStudentForm">
+       <div style="margin: 10px 0;">
+           <label for="studentName">Student Name</label>
+           <input type="text" id="studentName" required="true" placeholder="Enter Student Name" />
+       </div>
+       <input type="submit" value="Submit" onclick="saveTest()" />
+   </form>
+   <script src="/soap/ajax/45.0/connection.js"></script>
+   <script>
+       function saveTest()
```
Comment (meta-vijay-khatri) [inline JavaScript and Ajax/ClassDetailPage.page:14]: `Test` in method name ??

File: JavaScript and Ajax/ClassDetailPage.page [hunk 2]
```diff
@@ -0,0 +1,38 @@
+<apex:page standardController="Class__c">
```
Comment (meta-vijay-khatri) [inline JavaScript and Ajax/ClassDetailPage.page:1]: page name is in correct this page is used to create/add a student to the class

File: JavaScript and Ajax/DeleteStudentRecord.page [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+            <input type="button" onclick = "cancel()" value="cancel" />
+        </form>
+    </body>
+    <script src="/soap/ajax/45.0/connection.js"></script>
+    <script>
+        function deleteRecords()
+    {
+        event.preventDefault();
+        try{
+            const queryString = window.location.search;
```
Comment (meta-vijay-khatri) [inline JavaScript and Ajax/DeleteStudentRecord.page:17]: queryString ???

File: JavaScript and Ajax/DeleteStudentRecord.page [hunk 4]
```diff
@@ -0,0 +1,45 @@
+<apex:page >
+    <h1>
+        Do you want to delete these records
+    </h1>
+    <body>
+        <form style="margin: 20px;" id="Confirmation">
+            <input type="button" onclick = "deleteRecords()" value="Confirm" />
+            <input type="button" onclick = "cancel()" value="cancel" />
```
Comment (meta-vijay-khatri) [inline JavaScript and Ajax/DeleteStudentRecord.page:8]: there is no `cancel` function in this page...

File: JavaScript and Ajax/DeleteStudentRecord.page [hunk 5]
```diff
@@ -0,0 +1,45 @@
+<apex:page >
```
Comment (meta-vijay-khatri) [inline JavaScript and Ajax/DeleteStudentRecord.page:1]: there are unexpected errors on this page

Comment (meta-vijay-khatri) [inline JavaScript and Ajax/DeleteStudentRecord.page:1]: Some points from assignments are missing 

- why are we opening in new window?
- No records selected error message is missing.
- Cancel button does not work due to obvious reason (check comment on cancel method)

File: JavaScript and Ajax/StudentDeleteRemote.cls [hunk 6]
```diff
@@ -0,0 +1,19 @@
+public class StudentDeleteRemote {
+    
+    public List<Id> studentIdList { get; set; }
+    public StudentDeleteRemote() { } // empty constructor
+    
+    @RemoteAction
+    public static boolean deleteStudent(List<Id> studentIdList) {
```
Comment (meta-vijay-khatri) [inline JavaScript and Ajax/StudentDeleteRemote.cls:7]: `Boolean` data type should start with capital letter

File: JavaScript and Ajax/StudentDeleteRemote.page [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+    function deleteStudents() {
+        try{
+            const queryString = window.location.search;
+            const urlParams = new URLSearchParams(queryString);
+            var records = urlParams.getAll('id');
+            Visualforce.remoting.Manager.invokeAction(
+                '{!$RemoteAction.StudentDeleteRemote.deleteStudent}',
+                records, 
+                function(result, event){
+                    if (event.status) {
```
Comment (meta-vijay-khatri) [inline JavaScript and Ajax/StudentDeleteRemote.page:12]: what if the status code is 500 ??

**Review by meta-vijay-khatri — CHANGES_REQUESTED** [review]
It is appreciated if you add the metadata for the buttons as well.
few things I have noticed is that

- Button names are way incorrect.
- missing functionality at various places
- could not found the vf remote action assignment button on the org

File: JavaScript and Ajax/ClassDetailPage.page (hunk 1 above)
Comment (rathoreprem) [reply JavaScript and Ajax/ClassDetailPage.page:14]: changed it.

File: JavaScript and Ajax/ClassDetailPage.page (hunk 2 above)
Comment (rathoreprem) [reply JavaScript and Ajax/ClassDetailPage.page:1]: changed it.

File: JavaScript and Ajax/DeleteStudentRecord.page (hunk 3 above)
Comment (rathoreprem) [reply JavaScript and Ajax/DeleteStudentRecord.page:17]: changed it with another name...?

File: JavaScript and Ajax/DeleteStudentRecord.page (hunk 4 above)
Comment (rathoreprem) [reply JavaScript and Ajax/DeleteStudentRecord.page:8]: this page have a cancel function also.

Comment (rathoreprem) [reply JavaScript and Ajax/DeleteStudentRecord.page:8]: function*

File: JavaScript and Ajax/DeleteStudentRecord.page (hunk 5 above)
Comment (rathoreprem) [reply JavaScript and Ajax/DeleteStudentRecord.page:1]: Now on this page not unexpected error

File: JavaScript and Ajax/StudentDeleteRemote.cls (hunk 6 above)
Comment (rathoreprem) [reply JavaScript and Ajax/StudentDeleteRemote.cls:7]: changed it.

File: JavaScript and Ajax/StudentDeleteRemote.page (hunk 7 above)
Comment (rathoreprem) [reply JavaScript and Ajax/StudentDeleteRemote.page:12]: 500 Internal Server Error server error response code indicates that the server encountered an unexpected condition that prevented it from fulfilling the request.

**Pull request merged**